### PR TITLE
Adds a publishable dockerfile

### PR DESCRIPTION
**Pull request by evindunn**
##  Description
- Adds production.dockerfile which mimics the official conda installation process
- Updates README for installation via the resulting docker image
- Configured https://hub.docker.com/repository/docker/usgsastro/isis to build every time a new release is tagged in GH

## Related Issue
Not an official issue, but related to https://astrodiscuss.usgs.gov/t/missing-libarmadillo-when-installing-isis/504/4

## Motivation and Context
Simplified installation for those familiar with Docker containers

## How Has This Been Tested?
Built locally, ran `lowpass` on a cub

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by Kelvinrr — COMMENTED** [review]
Looks good to me but I have a testing concern of just making sure it works before pressing the button (anyone else have time for it? kinda swamped atm). And if this also needs an updated to the Jenkins pipeline to make sure this works?

File: docker/production.dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    conda config --add channels conda-forge && \
+    conda config --add channels usgs-astrogeology
+
+RUN conda install -c usgs-astrogeology isis
+
+RUN apt-get update && \
+    apt-get install libgl1 -y && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN bash -lc 'python "$CONDA_PREFIX/scripts/isisVarInit.py"'
```
Comment (jessemapel) [inline docker/production.dockerfile:16]: I am concerned about this part. The data area is not installed as part of the anaconda install process. The data area is also quite large and I'd be worried about downloading it inside of the docker container. So, how would users get the container using a local copy of the data area?

Comment (Kelvinrr) [reply docker/production.dockerfile:16]: Probably something that should be added to docs he included, but `isisVarInit.py` should just set the default envars and you would download DATA/TESTDATA on the host machine (not in the container) and do a `-v $ISISTESTDATA:/opt/conda/testData` or similar to mount your downloads into the container.

Comment (jessemapel) [reply docker/production.dockerfile:16]: @evindunn Can you add this to the installation via docker instructions?

**Review by jessemapel — DISMISSED** [review]
I think this looks good to me.

File: docker/production.dockerfile (hunk 1 above)
Comment (evindunn) [reply docker/production.dockerfile:16]: @jessemapel  @Kelvinrr how's [this](https://github.com/evindunn/ISIS3/tree/docker-image#installation-with-docker) ?

**Comment by Kelvinrr** [comment]
running through the instructions, after running `docker build -t isis -f production.dockerfile .` I get: 

```
Step 5/5 : RUN bash -lc 'python "$CONDA_PREFIX/scripts/isisVarInit.py"'
 ---> Running in 73b724f0f365
python: can't open file '/opt/conda/scripts/isisVarInit.py': [Errno 2] No such file or directory
The command '/bin/sh -c bash -lc 'python "$CONDA_PREFIX/scripts/isisVarInit.py"'' returned a non-zero code: 2
```

**Comment by evindunn** [comment]
@Kelvinrr weird, it builds on mine. I'll experiment some more.

**Comment by evindunn** [comment]
@Kelvinrr just tried again and it works on mine.

```console
$ docker build -t isis -f production.dockerfile .
...
 => [5/5] RUN bash -lc 'python "$CONDA_PREFIX/scripts/isisVarInit.py"'                                                                                                                  0.5s
 => exporting to image                                                                                                                                                                 16.6s 
 => => exporting layers                                                                                                                                                                16.5s 
 => => writing image sha256:d5034d4b3014e86e93d4f4ac6b473aebd55078f5acca2db8f1d82e199aab8eb8                                                                                            0.0s 
 => => naming to docker.io/library/isis
```

Only descrepency I could think of was [buildkit](https://docs.docker.com/develop/develop-images/build_enhancements/), which I have enabled by default.

```console
$ DOCKER_BUILDKIT=0 docker build -t isis -f production.dockerfile .
...
Step 5/5 : RUN bash -lc 'python "$CONDA_PREFIX/scripts/isisVarInit.py"'
 ---> Running in 308c9d92b898
python: can't open file '/opt/conda/scripts/isisVarInit.py': [Errno 2] No such file or directory
The command '/bin/sh -c bash -lc 'python "$CONDA_PREFIX/scripts/isisVarInit.py"'' returned a non-zero code: 2
```

Looks like the buildkit version retrieves isis 4.3.0 (isisVarInit.py), but the non-buildkit version retrieves isis 4.2.0 (isis3VarInit.py).

With buildkit
```
# conda list isis
# packages in environment at /opt/conda:
#
# Name                    Version                   Build  Channel
isis                      4.3.0                         0    usgs-astrogeology
```

Without buildkit
```
# conda list isis
# packages in environment at /opt/conda:
#
# Name                    Version                   Build  Channel
isis                      4.2.0                         0    usgs-astrogeology
```

Exact same Dockerfile, which is maddening.

**Review by Kelvinrr — DISMISSED** [review]

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**